### DISCUSSION TITLE
New user. Workflow for plugin updates?

### DISCUSSION THREAD

**Discussion by grimm26** [jdx/mise#1823] [Q&A]
I'm a new user and found mise as a nice way to maintain installation of various tools that may not have OS pkgs or at least not up to date ones. I Use yadm and have a bootstrap script that I use a few times week to keep everything up to date. I had previously just had a list of pkgs that I would check for latest version on github and then pull down and install if I wasn't up to date.

Now I have a list of mise plugins and am using mise but am trying to lockdown the procedure for updating everything. At this point I am just doing global versions of everything. It seems like this is what I need to do:

1. `mise self-update`
    - updates mise itself, and git pulls  plugin repos?
 2. run through my list of plugins and do a `mise use --global --yes $pkg`
    - Make sure all of the tools that I want are installed
 3. run a `mise upgrade --yes` to update all installed tools.

At first I thought self-update would actually update all of the installed tool versions, too, since the help for `self-update` says "By default, this will also update any installed plugins" but that seems not to be the case. It appears to just "update" the tool repos by doing a pull since for instance I have `awscli` set to `latest` and after a `mise self-update` it still shows up as outdated.

**Comment by Marukome0743** [comment]
These commands update three different layers:

1. **The mise executable**
   - Use the same package manager that installed mise, or `mise self-update` for a self-managed installation.
   - `mise self-update` also updates installed plugin repositories by default, but it does not upgrade installed tool versions.

2. **Plugin repositories**
   - `mise plugins update` updates all installed plugin repositories.
   - `mise plugins update <plugin>` updates only the specified plugin.
   - This updates the installation logic and version-listing code, not the installed tool version.

3. **Installed tool versions**
   - `mise install` makes sure the versions declared in the loaded configuration are installed.
   - `mise upgrade` upgrades configured tools within their existing version requests.
   - `mise upgrade --bump` may also update the version requests stored in `mise.toml`.

A typical workflow is therefore:

```sh
# Only for installations that support mise self-update:
mise self-update

# Optional when self-update already updated the plugins:
mise plugins update

# Ensure every configured tool is installed:
mise install

# Upgrade installed tools:
mise upgrade
```

Use `mise upgrade --bump` when you also want to move a configuration such as `node = "20"` to a newer major version. Without `--bump`, mise generally stays within the configured request.

You do not need to loop over the tools and call `mise use --global` each time. Declare the desired global tools in the global mise config and use `mise install`/`mise upgrade` to maintain them.

I verified the distinction against the current main branch with a clean plugin and tool installation. `mise plugins update tiny` updated the plugin repository without changing the installed `tiny@1.0.0`. A subsequent `mise upgrade --bump tiny` installed `tiny@3.1.0`, removed 1.0.0, and updated the config entry to `tiny = "3.1.0"`.

This is why `awscli` could still appear outdated after `mise self-update`: the plugin may have been updated, but upgrading the installed AWS CLI version is a separate `mise upgrade` operation.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
